aten/src/THCUNN/SpatialUpSamplingNearest.cu

```diff
@@ -2,85 +2,101 @@
 #include "common.h"
 #include "THCTensor.hpp"
 
-#include <thrust/transform.h>
-#include <thrust/reduce.h>
-#include <thrust/transform_reduce.h>
-#include <thrust/functional.h>
+#include "linear_upsampling.h"
+#include "THCDeviceTensor.cuh"
+#include "THCDeviceTensorUtils.cuh"
+#include "THCDeviceUtils.cuh"
 
 #include "THCHalf.h"
 #include "THCHalfAutoNumerics.cuh"
+#include "THCAtomics.cuh"
 
-/*
- * Description:
- */
+template<typename Dtype, typename Acctype>
+__global__ void nearest_neighbor_interp2_kernel(
```

Comment:
Why do we need new NN kernels? I remember that you briefly talked about it, but I didn't really understand.

Reply:
Do you mean we could reuse kernels for linear methods? I guess the main motivation for this was "we had a kernel method for each (linear/nearest , 1d/2d/3d) combination before" and then I thought we might have some reasons to keep specific kernels. But honestly the answer is i'm not sure :P

Reply:
Ah okay. I was just mostly wondering why we can't patch the old kernels to make them work.

Reply:
Yea I'm not sure about this neither. CC: @fmassa and see if he has any thoughts :D

Reply:
I feel it's fine if you want to change. I was just curious.